src/Cli/dotnet/Commands/Tool/Execute/ToolExecuteCommand.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.DotNet.Cli.CommandFactory;
+using Microsoft.DotNet.Cli.CommandFactory.CommandResolution;
+using Microsoft.DotNet.Cli.Commands.Tool.Install;
+using Microsoft.DotNet.Cli.ToolPackage;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.Commands.Tool.Execute
+{
+    internal class ToolExecuteCommand(ParseResult result) : CommandBase(result)
+    {
+        private readonly PackageIdentity? _packageToolIdentityArgument = result.GetValue(ToolExecuteCommandParser.PackageIdentityArgument);
+        private readonly IEnumerable<string> _forwardArguments = result.GetValue(ToolExecuteCommandParser.CommandArgument) ?? [];
+        private readonly bool _allowRollForward = result.GetValue(ToolExecuteCommandParser.RollForwardOption);
+        private readonly string? _configFile = result.GetValue(ToolExecuteCommandParser.ConfigOption);
+        private readonly string[] _sources = result.GetValue(ToolExecuteCommandParser.SourceOption) ?? [];
+        private readonly string[] _addSource = result.GetValue(ToolExecuteCommandParser.AddSourceOption) ?? [];
+        private readonly bool _ignoreFailedSources = result.GetValue(ToolCommandRestorePassThroughOptions.IgnoreFailedSourcesOption);
+        private readonly bool _interactive = result.GetValue(ToolExecuteCommandParser.InteractiveOption);
+        private readonly VerbosityOptions _verbosity = result.GetValue(ToolExecuteCommandParser.VerbosityOption);
+        private readonly bool _yes = result.GetValue(ToolExecuteCommandParser.YesOption);
+        private readonly bool _prerelease = result.GetValue(ToolExecuteCommandParser.PrereleaseOption);
+
+        public override int Execute()
+        {
+            if (!UserAgreedToRunFromSource() || _packageToolIdentityArgument is null)
```

Comment:
How does this show up to the user? Just as "Tool exec command returned 1"? If so, we should make that clearer.